src/axom/inlet/Field.cpp

```diff
@@ -58,12 +60,168 @@ bool Field::required()
                                   " value named {0}",
                                   m_sidreGroup->getName());
     SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
     return false;
   }
 
   return (bool)intValue;
 }
 
+std::shared_ptr<Field> Field::defaultValue(const char* value) {
+  std::string str = "";
+  if (value) {
+    str = value;
+  }
+  return defaultValue(str);
+}
+
+std::shared_ptr<Field> Field::defaultValue(const std::string& value) {
+  if (m_type != axom::sidre::DataTypeId::CHAR8_STR_ID) {
+    SLIC_WARNING("Field value type did not match STRING");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                  m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewString("defaultValue", value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewString("value", value);
+    }
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(bool value) {
+  if (m_type != axom::sidre::DataTypeId::INT8_ID) {
+    SLIC_WARNING("Field value type did not match BOOL");
+    setWarningFlag(m_sidreRootGroup);
+  }
+  if (m_sidreGroup->hasView("defaultValue")) {
+    std::string msg = fmt::format("Inlet Field has already defined default value: {0}",
+                                   m_sidreGroup->getName());
+    SLIC_WARNING(msg);
+    setWarningFlag(m_sidreRootGroup);
+  } else {
+    if (m_docEnabled) {
+      m_sidreGroup->createViewScalar("defaultValue", (int8)value);
+    }
+    if (!m_sidreGroup->hasView("value")) {
+      m_sidreGroup->createViewScalar("value", (int8)value);
+    }
+  }
+  return shared_from_this();
+}
+
+std::shared_ptr<Field> Field::defaultValue(int value) {
+  if (m_type != axom::sidre::DataTypeId::INT_ID) {
```

Comment:
Will this cause problems if the field is a double, but the user sets the default value using an int.
E.g.
```cpp
   addDouble("Table1/Table2/foo")->defaultValue(0.); //  OK
   addDouble("Table1/Table2/foo")->defaultValue(0);  //  WARNING
```

I think you'd only want to warn the user if it cannot be cast to the desired type.

Reply:
Please add an example like this to the unit tests.
Or create a ticket if it does not currently work as expected.